api/tools/src/main/java/org/pmiops/workbench/tools/BackfillGSuiteUserData.java

```diff
@@ -0,0 +1,131 @@
+package org.pmiops.workbench.tools;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.apache.ApacheHttpTransport;
+import com.google.gson.Gson;
+import org.pmiops.workbench.config.CacheSpringConfiguration;
+import org.pmiops.workbench.config.RetryConfig;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.ConfigDao;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.model.Config;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.google.DirectoryService;
+import org.pmiops.workbench.google.DirectoryServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import static org.springframework.context.annotation.FilterType.ASSIGNABLE_TYPE;
+
+/**
+ * See api/project.rb backfill-gsuite-fields for usage.
+ * <p>
+ * This command-line tool updates some GSuite data for all existing AoU users to match the data
+ * we now populate for new users.
+ *
+ * Once the backfill is completed for all environments, this script may be deleted.
+ *
+ * TODO(RW-2032): Before this script goes away, but after recoveryEmail becomes available
+ * to the directory API, we should augment this script & run a backfill for recovery emails.
+ */
+@SpringBootApplication
+// Most of the "config" module contains Spring beans that we don't want. But the RetryConfig
+// is required for the DirectoryService class, so we specifically include only that one.
+//
+// TODO(gjuggler): find a better way to manage component scans like this... either cenralize
```

Comment:
Probably we need to split out packages / folder structure and define clear relationships so we can easily pull in non-request-scoped stuff. I'm not sure whether that's something we can easily enforce with our current build tooling or if it would have to just be convention.